frontend/components/text2text/results/Text2TextList.vue

```diff
@@ -251,13 +252,37 @@ export default {
         ],
       });
     },
+    async toggleValidateRecord() {
+      if (this.record.status === "Validated") {
+        await this.onChangeStatusToDefault();
+      } else {
+        this.onValidate();
+      }
+    },
+    async toggleDiscardRecord() {
+      if (this.record.status === "Discarded") {
+        await this.onChangeStatusToDefault();
+      } else {
+        this.onDiscard();
+      }
+    },
     async onValidate() {
       let newS = {
         score: 1,
         text: this.record.sentenceForAnnotation || null,
       };
       this.$emit("annotate", { sentences: [newS] });
     },
+    async onChangeStatusToDefault() {
+      await this.changeStatusToDefault({
+        dataset: this.getText2TextDataset(),
+        records: [
+          {
+            ...this.record,
+          },
+        ],
+      });
+    },
```

Comment:
1/ add a factory function before passing it to the `changeStatusToDefault` function
2/ the destructuration `{... this.record}` seems better to be replaced directly by `this.record`